### ISSUE TITLE
When using TypeScript 4.x, live-reload is much slower

### ISSUE THREAD

**Issue by boris-petrov** [typed-ember/ember-cli-typescript#1278]
Some context could be found in [Discord](https://discord.com/channels/480462759797063690/484421406659182603/747723822681161748). TLDR is `tsc --noEmit --watch` finishes immediately, but `ember serve` takes a few seconds when a TS file changes. Running it with `DEBUG=ember-cli-typescript*` shows:

```
file changed helpers/create-immediate-value.ts
⠙ building...   ember-cli-typescript:typecheck-worker Typecheck starting +339ms
⠏ building...   ember-cli-typescript:typecheck-worker Typecheck complete (0 diagnostics) +5s
```

cc @dfreeman

**Comment by chriskrycho** [comment]
Thank you for reporting!

**Comment by NullVoxPopuli** [comment]
Related: https://github.com/typed-ember/ember-cli-typescript/issues/1321

**Comment by andreyfel** [comment]
Hey guys! Any update on this? I also noticed the slowdown, so, decided to stay on TS@3.9 for now.
It is reproducible with e-c-t@4 as well.

**Comment by NullVoxPopuli** [comment]
In my own testing, I've noticed some slow down in Babel as well. :( 

Swc can't come soon enough. :D

**Comment by dfreeman** [comment]
@NullVoxPopuli Neither `typescript@4` nor `ember-cli-typescript@4` should impact your Babel compilation speed. The `typescript` package itself doesn't play a role in emit, and for `ember-cli-typescript`, the actual Babel plugins that are being set up didn't change between v3 and v4, just which package that responsibility lies with.

**Comment by dfreeman** [comment]
I spent a little time with this this afternoon, and the performance characteristics of the compiler API we use to capture typechecking results seem to have changed in 4.0. Specifically, the version of that API that only tracks diagnostics has become much slower than the version that also tracks emitting files, which is a bit surprising to me.

I opened https://github.com/microsoft/TypeScript/issues/40808, and we'll see if this is a bug and figure out where to go from there.
